// data-analysis-python\src\generate_chart_lc_sim\generate_chart_lc_sim.proto

syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/generate_chart_lc_sim";

// GenerateChartLCServiceというサービスを定義
service GenerateChartLCService {
  rpc GenerateChart (GenerateChartLCRequest) returns (GenerateChartLCResponse);
}

// リクエストメッセージを定義
message GenerateChartLCRequest {
  repeated string dates = 1;    // 日付のリスト
  repeated double close_prices = 2; // 株価のリスト
  string purchase_date = 3;     // 購入日
  double purchase_price = 4;    // 購入価格
  string end_date = 5;          // 売却日
  double end_price = 6;         // 売却価格
}

// レスポンスメッセージを定義
message GenerateChartLCResponse {
  string chart_data = 1; // チャート可視化データ (Base64エンコード) を含むフィールド
  bool success = 2;      // プロット生成の成否
  string message = 3;    // メッセージ（エラーメッセージ等）
}



// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/generate_chart_lc_sim --python_out=src/generate_chart_lc_sim --grpc_python_out=src/generate_chart_lc_sim src/generate_chart_lc_sim/generate_chart_lc_sim.proto


// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/generate_chart_lc_sim --go_out=./src/service/ms_gateway/generate_chart_lc_sim --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/generate_chart_lc_sim --go-grpc_opt=paths=source_relative ../data-analysis-python/src/generate_chart_lc_sim/generate_chart_lc_sim.proto
